power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -329,15 +329,15 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         is_asym_parameter_up_to_date_ = is_asym_parameter_up_to_date_ && !changes.topo && !changes.param;
     }
 
-    template <math_output_type MathOutputType, typename MathSolverType, typename YBus, typename InputType,
+    template <solver_output_type SolverOutputType, typename MathSolverType, typename YBus, typename InputType,
```

Comment:
most of the objects in this file are actually `MathOutput` types (in the sense that it should be equal to the output after the optimizer is done with it).

in particular, anywhere where it doesn't matter whether or not it is actually a `std::vector` of `SolverOutput`s is a location where it should probably remain `MathOutput`

It might be a good idea to declare a `template <symmetry_tag sym> using MathOutput = std::vector<SolverOutput<sym>>;` in the `calculation_parameters.hpp`.

Reply:
I believe this should be part of the next PR?